services/identity/SitesService.ts

```diff
@@ -1,43 +1,104 @@
-import { ModelStatic } from "sequelize"
+import { ModelStatic, ValidationError } from "sequelize"
 
-import { Site } from "@database/models"
+import logger from "@logger/logger"
+
+import { BadRequestError } from "@errors/BadRequestError"
 
-import TokenStore from "./TokenStore"
+import { Site } from "@database/models"
 
-interface SitesServiceProps {
+export interface SitesServiceProps {
   repository: ModelStatic<Site>
-  tokenStore: TokenStore
+}
+
+export interface CreateSiteProps {
+  repositoryName: string
+  createdBy?: string
+  agency?: string
+  siteName?: string
+  contact?: string
+  repositoryUrl?: string
+  hostingId?: string
+  stagingUrl?: string
+  productionUrl?: string
+  liveDomain?: string
+  redirectFrom?: string[]
+  uptimeId?: string
+  uptimeUrl?: string
+  launchedAt?: string
+  launchedBy?: string
+}
```

Comment:
this interface is huge and has a few un-necessary stuff (we should only require very basic info). this is a direct consequence of our data model in the above and we won't ever need alot of stuff here (eg: prod/staging url because we will assign it when we create ._.)